shared/src/business/utilities/DateHandler.js

```diff
@@ -37,7 +37,13 @@ const prepareDateFromString = (dateString, inputFormat) => {
   if (dateString === undefined) {
     dateString = createISODateString();
   }
-  return moment.tz(dateString, inputFormat, USTC_TZ);
+  const dateFromMoment = moment.tz(dateString, inputFormat, USTC_TZ);
+  // If an invalid string is passed in, we're able to check the result and log so error is
+  // not silently swallowed
+  if (dateString != null && !dateFromMoment.isValid()) {
+    console.error('Invalid date', dateString);
```

Comment:
How is this error being thrown?

Reply:
Simply logged for archeological purposes, with a stack trace to pinpoint the issue since it can be a little convoluted

Reply:
Yeah, bigger question was how/whether we wanted to capture something in the logs that we were receiving a string that can't possibly be a date.

Reply:
1) I don't like invoking "null" almost ever. (I personally wish it did not exist in JS).  Is there any evidence that it's ever being provided as the value of the first argument?
2) if you want to simply note the issue but not create big traces, I think perhaps `applicationContext.logger.warn(<string>)` would be a better choice.

Reply:
Kris, I also considered that, but we _currently_ do not pass in applicationContext here 😬

Reply:
We do pass null in several times, as several functions in dateHandler itself return null. The only important change is the one on lines 301 to 305, which prevents the invalid `moment` object being passed _back_ into moment in `formatDateString`, eliminating a lot of deprecation warnings. The rest is an attempt to log when a bad string is passed into moment, because it happens often and is being swallowed silently.